src/pveclib/vec_char_ppc.h

```diff
@@ -137,6 +137,13 @@
  */
 
 ///@cond INTERNAL
+#ifndef vec_popcntb
+static inline vui8_t vec_popcntb (vui8_t vra);
+#else
+/* Work around for GCC PR85830.  */
+#undef vec_popcntb
+#define vec_popcntb __builtin_vec_vpopcntb
+#endif
```

Comment:
not really. This is C so no overloading. And vec_popcntb is a macro defined (sometime incorrectly, See GCC PR85830) in <altivec.h>. So all I can do is define the inline function only if the macro it not defined. If defined make sure it is defined correctly.

Reply:
Well, these intrinsics are weird, and I think GCC supports overloading with respect to types even in C for them. That's been my experience, and what is documented in the _Power Vector Intrinsics Programming Reference_.

Reply:
No. GCC for POWER supports type generic built-ins for altivec.h using internal GCC mechanisms. Clang only supports the type generic built-ins. Plus PVECLIB has to support the operation even if the -mcpu target does not. The math.h type generic macros are a freaking nightmare of complexity which I have no plans to emulate. I am not ready to entertain a C++ template approach as that would cause my head to explode and would delay P10 support indefinitely.

If you want to address type generics for PVECLIB as a layer on top, please provide a detailed proposal. That you personally have time to work on ;)